### DISCUSSION TITLE
GEN24_Ladesteuerung v0.12.2

### DISCUSSION THREAD

**Discussion by wiggal** [wiggal/GEN24_Ladesteuerung#45] [Announcements]
Screenshots aktualisiert und Readme angepasst

Änderungen in html/7_tab_Diagram.php und html/funktion_Diagram.php
- Footer: Element "bis ..." weglassen wen von und bis gleich.
- Tooltip: Summenbildung hinzugefügt

<hr /><em>This discussion was created from the release <a href='https://github.com/wiggal/GEN24_Ladesteuerung/releases/tag/v0.12.2'>GEN24_Ladesteuerung v0.12.2</a>.</em>

**Comment by jaydee73** [comment]
Moin Wiggal,

ich bereite gerade vor, von meiner Version 0.11.1 auf die 0.12.2 upzudaten. Bei mir als Docker-Version.

Du hast ja Logging via SQLite-DB eingeführt. Wo liegt denn die DB-Datei? Ich würde die natürlich gerne via Mapping in den Container legen, damit sie auch bei einem Container-Upate nicht verloren geht. Das wird ja aufgrund der Docker-Architektur sonst nicht anders gehen...

Im Source-Code habe ich keine SQLite-DB-Datei gefunden. Wird die neu angelegt?

**Reply by wiggal** [reply]
Servus,
habe ich eigentlich für die Dockerversion so vorbereitet, ich hoffe es funktioniert.
Siehe Beschreibung:
https://github.com/wiggal/Docker_gen24_ladesteuerung/wiki/Installation-Dockerimage-von-Gen24%E2%80%90Ladesteuerung-im-Synology%E2%80%90Docker

**Reply by jaydee73** [reply]
Hach wie peinlich. Hab natürlich meine eigene Doku nicht vorher angeschaut. ;-)

**Reply by jaydee73** [reply]
Ich sehe aber gerade, hier hast du noch einen kleinen Fehler drin in der Doku:

https://github.com/wiggal/Docker_gen24_ladesteuerung

**schnipp**
NEU ab Version 0.12.0
- ./CONFIGS/**Crontab.log**:/home/GEN24/PV_Daten.sqlite
ENDE NEU

...
- ./CONFIGS/**Crontab.log**:/home/GEN24/PV_Daten.sqlite

...

**schnapp**

Die von mir fett markierten "Crontab.log" müsste "PV_Daten.sqlite" heißen, oder?

**Reply by wiggal** [reply]
Ja natürlich, habs berichtigt.
Sorry

**Comment by jaydee73** [comment]
Ok, erste Hürde in der 0.12.2:

Du hast ja das Feature "zusätzliche Ladeblöcke" eingeführt, welches ich grundsätzlich großartig finde. Zu Beginn wollte ich dieses Feature aber erstmal "deaktivieren", also nur die Standard-Ladeberechnung nehmen. Also habe ich alle zum Thema Ladeblöcke auskommentiert. Daraufhin kam dann folgendes im Log:

`ERROR: die Variable [Ladeberechnung][Zusatz_Ladebloecke] wurde NICHT definiert! `

Gut, dachte ich mir, gebe ich ihm zumindest diese Zeile und dann zwei Ladeblöcke, in denen ich aber nichts definiere. Dann kommt folgendes:

`ERROR: die Variable [Winter][Monate] wurde NICHT definiert! `

Er will also anscheinend zwangsläufig da was haben. Eine Möglichkeit, die weiteren Ladeblöcke (vorerst) nicht zu nutzen, gibt es also nicht? Das Einfachste wäre ja, da via C&P die Einträge aus der normalen Ladeberechnung reinzukopieren und dann irgendwelche Monate zu wählen (ist ja egal, da überall gleiche Werte). Oder wie würdest du das machen?

**Reply by wiggal** [reply]
ja gut, hab ich wohl nicht angefangen.
es müsste ein Block, mit einem Monat und einer Variablen, reichen, hoffe ich.
Der Wert der Variablen kann auch gleich sein.
Ich werd es mit der nächsten Version abfangen,
dass es keine Zusatzblöcke braucht.

**Reply by jaydee73** [reply]
Ok, keine Eile. Habe jetzt Blöcke mit identischen Wert-Variablen und deinen vordefinierten Monaten genommen. Nun geht es. Bzw. ich glaube, es geht... ;-) Crontab.log sieht gut aus. Werte sind aber im Diagramm noch nicht zu sehen. Liegt vielleicht auch daran, dass mein WR derzeit schläft...

Ach so, ein kleiner Feature-Wunsch vielleicht noch: Könntest du irgendwo (auf der Webserver-Hilfe-Seite evtl.?) noch die Versionsnummer der aktuell installierten Version einblenden? Könnte man ja aus dem Changelog parsen oder so...

**Reply by jaydee73** [reply]
Update: Ah, doch, jetzt sehe ich die ersten Werte...

**Reply by wiggal** [reply]
wegen der Versionsnummer will ich nicht extra eine Datei lesen. 
Bin noch am überlegen, evtl. fest in die Hilfedatei reinschreiben, ist halt schnell übersehen.
🤔

**Reply by wiggal** [reply]
Hab die Versionsnummer nun fest in die Hilfedatei eingebaut.
Mal sehen wie oft ich vergess sie zu ändern ;-)

**Reply by jaydee73** [reply]
Deswegen hatte ich halt überlegt, wo sie noch steht und eben nicht manuell gepflegt werden muss. Da fiel mir nur das Changelog ein.
